### DISCUSSION TITLE
vips left many tmp fd files "/tmp/vips-117-385983422.v (deleted)"

### DISCUSSION THREAD

**Discussion by t00350320** [libvips/libvips#3770] [Q&A]
## Bug report

**Describe the bug**
vips left many tmp fd files "/tmp/vips-117-385983422.v (deleted)"

```
lsof | grep delete
image-xx  1346 11422 image-xx service   40u      REG             253,46 547345798     153605 /tmp/vips-48-1620176665.v (deleted)
image-xx  1346 11422 image-xx service   41u      REG             253,46 547345798     153602 /tmp/vips-45-1163964295.v (deleted)
image-xx  1346 11422 image-xx service   42u      REG             253,46 547345798     153603 /tmp/vips-46-2261230589.v (deleted)
image-xx  1346 11422 image-xx service   43u      REG             253,46 547345798     153609 /tmp/vips-112-48929434.v (deleted)
image-xx  1346 11422 image-xx service   44u      REG             253,46 547345798     153607 /tmp/vips-110-2445733788.v (deleted)
image-xx  1346 11422 image-xx service   46u      REG             253,46 547345798     153612 /tmp/vips-118-2428124265.v (deleted)
image-xx  1346 11422 image-xx service   48u      REG             253,46 547345798     153611 /tmp/vips-117-385983422.v (deleted)
root@10-145-189-110:~# vips -v
vips-8.15.0
root@10-145-189-110:~# lsof | grep delete | wc -l
959

/proc/1346/fd
lrwx------ 1 service service 64 Nov 30 10:07 40 -> '/tmp/vips-48-1620176665.v (deleted)'
lrwx------ 1 service service 64 Nov 30 10:07 41 -> '/tmp/vips-45-1163964295.v (deleted)'
lrwx------ 1 service service 64 Nov 30 10:07 42 -> '/tmp/vips-46-2261230589.v (deleted)'
lrwx------ 1 service service 64 Nov 30 10:07 43 -> '/tmp/vips-112-48929434.v (deleted)'
lrwx------ 1 service service 64 Nov 30 10:07 44 -> '/tmp/vips-110-2445733788.v (deleted)'
lrwx------ 1 service service 64 Nov 30 10:07 46 -> '/tmp/vips-118-2428124265.v (deleted)'
lrwx------ 1 service service 64 Nov 30 10:07 47 -> 'socket:[3533616797]'
lrwx------ 1 service service 64 Nov 30 10:07 48 -> '/tmp/vips-117-385983422.v (deleted)'
```

**To Reproduce**
unconfirmed

**Expected behavior**
deleted files should be cleaned up absolutely

**Environment**
(please complete the following information)
 - OS: [Linux version 3.10.0-957.27.2.el7.x86_64]
 - Vips: [vips-8.15.0]

thanks !!!

**Comment by jcupitt** [comment]
Hi @t00350320,

I think you're seeing the libvips operation cache. Did you see the section in the docs about opening files?

https://www.libvips.org/API/current/How-it-opens-files.html

If you open a large file (over 100mb when decompressed) for random access, libvips will keep a decompressed version in a temporary disc file, and reuse it if you open the same file again. You can control the cache size with these functions:

https://www.libvips.org/API/current/VipsOperation.html#vips-cache-set-max

**Reply by t00350320** [reply]
Because our prod servers' current disk usage is too high.
If we do not want to cache the transcoded files, and want to delete it immediately , how can we deal with this case. 
In orther words, how to balance the  disk  and memory usage?

**Reply by jcupitt** [reply]
You probably shouldn't be using random access mode anyway, so I suspect there are some issues in the code you're using.

What kind of processing are you doing? Can you share any source code?
